### PR TITLE
[102X] update JetID cuts

### PR DESCRIPTION
**Pull request by karavdin**
Add JetID for 2016/2017/2018 together, set up working points for CHS and PUPPI
[onlycompile]

### CODE REVIEW

File: common/src/JetIds.cxx [hunk 1]
```diff
@@ -53,54 +53,229 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
...
+    if(fabs(jet.eta())>=2.4)
+      return true;
+      
+    if(jet.chargedEmEnergyFraction()<0.99
+       && jet.chargedHadronEnergyFraction()>0
+       && jet.chargedMultiplicity()>0)
+      return true;   
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.90
```
Comment (raggleton) [inline common/src/JetIds.cxx]: Hm looks like they updated the requirements for eta 2.7 - 3, now it's `jet.neutralEmEnergyFraction()>0.01 && jet.neutralHadronEnergyFraction()<0.98` instead.

File: common/src/JetIds.cxx [hunk 2]
```diff
@@ -53,54 +53,229 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
...
+  return tightID2016_CHS(jet);
+}
+
+
+
+bool JetPFID::tightLepVetoID2016(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;//no requirement for eta>2.7
+  if(!tightID2016_CHS(jet)) return false;
+  if(jet.muonEnergyFraction()>=0.8) return false;
+  if(fabs(jet.eta())<=2.4 && jet.chargedEmEnergyFraction()<0.90) return false;
```
Comment (raggleton) [inline common/src/JetIds.cxx]: chargedEM inequality needs reversing direction

File: common/src/JetIds.cxx [hunk 3]
```diff
@@ -53,54 +53,229 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
...
+
+bool JetPFID::tightID2016_PUPPI(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;
+  return tightID2016_CHS(jet);
+}
+
+
+
+bool JetPFID::tightLepVetoID2016(const Jet & jet) const{
+  if(fabs(jet.eta())>2.7) return true;//no requirement for eta>2.7
```
Comment (raggleton) [inline common/src/JetIds.cxx]: I never know whether they don't put any requirement on the twiki page on the assumption people just revert to the normal tight ID? i.e. would it be better to do here `if(fabs(jet.eta())>2.7) return tightID2016_CHS(jet);` ?

Comment (karavdin) [reply common/src/JetIds.cxx]: This is the problem, not clear if default should be treated as CHS or as PUPPI

**Comment by raggleton** [comment]
Failure log:

```
test/test_JER.cpp: In constructor 'test_JER::test_JER(uhh2::Context&)':
test/test_JER.cpp:51:38: error: 'WP_TIGHT' is not a member of 'JetPFID'
   const JetId jetID(JetPFID(JetPFID::WP_TIGHT));
                                      ^~~~~~~~
creating obj/test/TestJetCorrections.o
make[1]: *** [../Makefile.common:101: obj/test/test_JER.o] Error 1
make[1]: *** Waiting for unfinished jobs....
test/test_JLCkey.cpp: In constructor 'test_JLCkey::test_JLCkey(uhh2::Context&)':
test/test_JLCkey.cpp:55:38: error: 'WP_TIGHT' is not a member of 'JetPFID'
   const JetId jetID(JetPFID(JetPFID::WP_TIGHT));
                                      ^~~~~~~~
```

File: common/src/JetIds.cxx [hunk 4]
```diff
@@ -53,54 +53,231 @@ bool DeepCSVBTag::operator()(const Jet & jet, const Event &) const{
...
+     && jet.chargedMultiplicity()>0){
+  }
+  else if(fabs(jet.eta())>2.6 && fabs(jet.eta())<=2.7
+	  && jet.neutralHadronEnergyFraction()<0.90 
+	  && jet.neutralEmEnergyFraction()<0.99
+	  && jet.chargedMultiplicity()>0){
+    return true;
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  && jet.neutralEmEnergyFraction()>0.99
```
Comment (raggleton) [inline common/src/JetIds.cxx]: Ah this one should be the other way round

**Comment by karavdin** [comment]
For some reason issues in test modules were not visible. Hopefully after "make clean, make" all issues with compilation are caught.

**Pull request merged**